app/actors/hyrax/actors/file_set_actor.rb

```diff
@@ -87,7 +87,7 @@ def attach_to_work(work, file_set_params = {})
       def attach_to_valkyrie_work(work, file_set_params)
         work = Hyrax.query_service.find_by(id: work.id) unless work.new_record
         file_set.visibility = work.visibility unless assign_visibility?(file_set_params)
-        Hyrax.persister.save(resource: file_set)
+        @file_set = Hyrax.persister.save(resource: file_set)
```

Comment:
can you say more about this instance variable?

it will be cached with this actor in the `ActionDispatch::MiddlewareStack` pool, which will hold it in memory until the next time its set. this can lead to unexpected behavior. see #3722